lib/platform/github/index.js

```diff
@@ -54,15 +55,11 @@ module.exports = {
 // Get all repositories that the user has access to
 async function getRepos(token, endpoint) {
   logger.info('Autodiscovering GitHub repositories');
-  logger.debug('getRepos(token, endpoint)');
-  if (token) {
-    process.env.GITHUB_TOKEN = token;
-  } else if (!process.env.GITHUB_TOKEN) {
+  const opts = endpoints.find({ platform: 'github' }, { token, endpoint });
+  if (!opts.token) {
     throw new Error('No token found for getRepos');
   }
-  if (endpoint) {
-    process.env.GITHUB_ENDPOINT = endpoint;
-  }
+  endpoints.update({ ...opts, platform: 'github', default: true });
```

Comment:
I'm wondering if there's a way to make the syntax more obvious about what it's doing. If not then it needs a comment added to explain